app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/supportingevidence/SupportingEvidenceController.scala

```diff
@@ -98,18 +98,42 @@ class SupportingEvidenceController @Inject() (
       case _ => Redirect(baseRoutes.StartController.start())
     }
 
-  def uploadSupportingEvidence(): Action[AnyContent] =
+  def uploadSupportingEvidence(isAmend: Boolean): Action[AnyContent] =
     authenticatedActionWithSessionData.async { implicit request =>
-      withUploadSupportingEvidenceAnswers { (_, _, answers) =>
+      withUploadSupportingEvidenceAnswers { (_, fillingOutClaim, answers) =>
         if (answers.fold(_.evidences, _.evidences).length >= maxUploads)
-          Redirect(routes.SupportingEvidenceController.checkYourAnswers())
-        else
-          upscanService
-            .initiate(
-              routes.SupportingEvidenceController
-                .handleUpscanErrorRedirect(),
-              routes.SupportingEvidenceController.scanProgress
+          Redirect(routes.SupportingEvidenceController.checkYourAnswers(isAmend))
+        else {
+          def purgeSession() = {
```

Comment:
can you explain this change?

Reply:
What exactly you don't understand here?

Reply:
it would be better named `changeAnswersState` - purge means delete

Reply:
ok, will change this